docs/performance/memory.txt

```diff
@@ -4,15 +4,40 @@
 Memory configuration
 ====================
 
-In general, you should allocate as much memory as possible to CrateDB.
+CrateDB is a Java application running on top of a Java Virtual Machine (JVM).
+For optimal performance you must configure the amount of memory that is
+available to the JVM for **heap** allocations. The **heap** is a memory region
+used for allocations of objects. For example, if you invoke a ``SELECT``
+statement, parts of the result set are temporarily allocated in the **heap**.
 
-The amount of memory made allocated is called the *heap*.
+The amount of memory that CrateDB can use for heap allocations is set using the
+`CRATE_HEAP_SIZE`_ environment variable.
 
-If the heap is too small, some memory-intensive queries (such as those which
-need to hold many rows in memory) may fail with ``OutOfMemory`` exceptions.
+The right memory configuration depends on your workloads. Consider the
+following when determining the right value:
```

Comment:
I would maybe pull the "Consider..." sentence into a new para, and then, because you're using a colon to introduce a list, I would format the next few paras as an unordered list

I would end the list with the last item being "CrateDB also uses Memory mapped files..."

Reply:
Pushed a fixup with the changes, but I'm not sure if I understood you right :)